### PR TITLE
Updates to Index page for data table + data dictionary for store refactor

### LINKED ISSUE #314: Adapt data dictionary and data table upload/store functionality for refactor on Index page
Code for the data dictionary found in `index.js` and `index.vue` now needs to be connected for the initial refactor work and to align with how the data table is set loaded – specifically that the data table must be uploaded first before data dictionary upload is allowed.

*Store*
- [ ] `processDataDictionary` action now calls mutation `setDataDictionary` with an object as payload containing the new data dictionary and the current data table's column names

*Index.vue*
- [ ] `processDataTable` and `processDataDictionary` actions are now called directly after a `file-selector` component emits that a file has been selected
- [ ] The file selector component for the data dictionary is now disabled until a data table has been chosen (new `file-selector` prop `enabled` is tied to the length of the data table in the store. See `dataTableSelected` computed value.)

*file-selector.vue*
- [ ] Ability to be disabled by parent (This stems from the need for a data dictionary not to be uploaded before a data table is first uploaded on the index page.) New CSS class `disabled-file-selector-button` has been added for this purpose and is swapped out via the new `labelClass` computed value.

### PR DESCRIPTION
**Pull request by jarmoza**
This closes #314 and #320 

Several updates here are made for the index page relating to the recent store refactor.

1. The workflow for the user has changed in that they are no longer able to select a data dictionary without first having selected a data table. This has been accomplished by adding a new feature to `file-selector` care of the prop `enabled`. File selection for a data dictionary is disabled until a data table has been selected. 
2. `textarea` elements on the index page have been converted to one way binding (user input does not edit underlying value) via the `value` attribute instead of the two-way bind via `v-model`
3. Reactivity for the textarea for the data dictionary input has been fixed by reassigning the data dictionary to itself in the store's `setDataTable` mutation once any new keys and values have been added or removed from the current existing dictionary.

### CODE REVIEW

**Comment by jarmoza** [comment]
@surchs Two further notes. 

1. Cypress component tests are all passing except the index page one. Will address this in the upcoming PR for adjusting tests (#309, #310, #311)
2. Can you provide any commentary necessary feedback how this branch was created/merged. My workflow was to branch off of https://github.com/neurobagel/annotation_tool/tree/jarmoza-315 to continue development while awaiting feedback for its PR, then to merge changes to the dev branch into this one before proceeding with its own PR submission.

File: pages/index.vue [hunk 1]
```diff
@@ -93,9 +95,15 @@
                 "dataTable"
             ]),
 
+            dataTableSelected() {
+
```
Comment (surchs) [inline pages/index.vue:99]: I think this guy may benefit from a small comment explaining why it is here. I.e.: checking if dataTable is already there. For future us mainly

File: store/index.js [hunk 2]
```diff
@@ -281,6 +280,10 @@ export const mutations = {
                               p_state.dataDictionary.annotated[column],
                               newDataDictionary[column]);
         }
+
+        // 2. Create a new object in case additions/deletions to the data
+        // dictionary object in order to maintain Vue reactivity
+        p_state.dataDictionary = Object.assign({}, p_state.dataDictionary);
```
Comment (surchs) [inline store/index.js:286]: :tada:

File: components/tool-navbar.vue [hunk 3]
```diff
@@ -24,7 +24,7 @@
                         :active="currentPageName === navItem.fullName"
                         :class="getNavItemColor(navItem)"
                         :data-cy="'menu-item-' + navItem.pageName"
-                        :disabled="!isPageAccessible(navItem.pageName)"
+                        :disabled="!isPageAccessible(pageData[getNextPage].pageName)"
```
Comment (surchs) [inline components/tool-navbar.vue:27]: Was this done by accident? This doesn't seem right.

**Review by surchs — CHANGES_REQUESTED** [review]
Thanks for the PR @jarmoza. Changes look good, take a look at the tool-nav change, it doesn't look right to me.

File: components/tool-navbar.vue (hunk 3 above)
Comment (jarmoza) [reply components/tool-navbar.vue:27]: Thanks for catching this. There was something funky that happened when I resolved merge conflict for this branch (thus my request for feedback in my second comment on this PR). Fixed.

Comment (surchs) [reply components/tool-navbar.vue:27]: Oh, sorry, didn't see your comments. Looking now!

**Comment by surchs** [comment]
> @surchs Two further notes.
> 
>     1. Cypress component tests are all passing except the index page one. Will address this in the upcoming PR for adjusting tests (resolving [refactor page test setup #309](https://github.com/neurobagel/annotation_tool/issues/309), [adjust index page tests #310](https://github.com/neurobagel/annotation_tool/issues/310), [adjust categorization page tests #311](https://github.com/neurobagel/annotation_tool/issues/311))

Sounds good. If you know why the index page one is currently failing, I'd include the fix here just so we keep the component tests green. If it's more involved, then let's do the fix with the e2e ones (that's how I undersand your comment).

>     2. Can you provide any feedback as to how this branch was created/merged? My workflow was to branch off of https://github.com/neurobagel/annotation_tool/tree/jarmoza-315 to continue development while awaiting feedback for its PR, then to merge changes to the dev branch into this one before proceeding with its own PR submission.

I am not super clear what has happened here. My guess would be that you did something like this:

1. Merge the previous PR (jarmoza-315) into dev
2. Merge your local copy of dev into jarmoza-314 (here) before pulling the new merge commit now on origin/dev
3. Then adding the missing changes from the merged PR into the already created local branch manually
4. possibly missing something in step 3.

Does any of this sound plausible? We could also chat about it more on Monday with the rest of the team, I think we haven't really finished discussing our GH workflow last time.

**Comment by jarmoza** [comment]
@surchs

1. I'll take a look at the e2e test again
2. That sounds plausible, although re: 2, I was almost positive I had pulled down the changes to the remote dev branch.

In any case, I wanted to see how this workflow would work. Good practice.

**Comment by surchs** [comment]
@jarmoza: I took another look because I wasn't really sure either. You are right, you had pulled the recent PR in the dev branch. I think what may have happened is that when you merged dev into jarmoza-314 there was a merge conflict because you might have already made changes to the same line (in `tool-navbar.vue`) on both jarmoza-314 and on the dev_... branch. So my best guess is that you accidentally kept the conflicting change from jarmoza-314 and dismissed the one from the dev branch, thus reverting this line to what it was before the merge.

It's tricky - good thing for us all to discuss again I think - I would also like to get better intuitions when reading a tricky log.

**Comment by jarmoza** [comment]
Okay, @surchs. We're good to go with the index page component test now. The data dictionary loading test needed to be updated. And while I was determining what was wrong with that test by comparing the code in the other tests, I went ahead and cleaned up the file - generalized `beforeEach` input data, removed extraneous setup data, and some formatting.

**Review by surchs — APPROVED** [review]
Looks great @jarmoza. I left a comment for a left-over thing in the tool-navbar that should be removed. Agreed, this is ready to merge! :tada:

**Pull request merged**